homeassistant/components/remote/harmony.py

```diff
@@ -215,10 +218,14 @@ def send_command(self, commands, **kwargs):
 
         num_repeats = kwargs.get(ATTR_NUM_REPEATS)
         delay_secs = kwargs.get(ATTR_DELAY_SECS, self._delay_secs)
+        hold_secs = kwargs.get(ATTR_HOLD_SECS)
+        _LOGGER.debug("Sending commands to device %s holding for %s seconds",
+                      device, hold_secs)
 
         for _ in range(num_repeats):
             for command in commands:
-                self._client.send_command(device, command)
+                _LOGGER.debug("Sending command %s", command)
+                self._client.send_command(device, command, hold_secs)
```

Comment:
We're not allowed to sleep in our thread pool. The delay introduced in pyharmony uses time.sleep. This isn't allowed. Either sleep in a separate thread or better move to asyncio.

Reply:
@MartinHjelmare, harmony is not async but sync. I understand that when using async there should be no blocking items (i.e. sleep, network, ...) but as this is still using sync and thus being handled by a SyncWorker and not in the event loop and thus per my understanding not in the main thread either.

Reply:
@MartinHjelmare Can you check my other comment. If I switch this all to async and then use hass.async_add_executor to call the API for Harmony (everything with self._client) would that work? Based on my understanding that is specifically there to be able to call sync methods and making HASS async.

Thx!

Reply:
No, it won't work as long as pyharmony uses `time.sleep`. Again, we are not allowed to sleep in our thread pool, which is what is used when scheduling a call with `hass.async_add_executor`.

The thread pool only has so many threads. If all threads of the thread pool would be sleeping at the same time, the pool would be exhausted and further work would be blocked. This is why we don't allow sleep in the thread pool.

Does it make sense?

Reply:
That makes sense. Thus now to get this moving forward I would need to makes this async, and then do the send_command in a separate thread created from harmony.py?

Changing pyharmony to async would be a potential breaking change for anyone else using it I would think.